crates/nargo_cli/src/errors.rs

```diff
@@ -45,7 +45,7 @@ pub(crate) enum CliError<B: Backend> {
 
     /// Error while compiling Noir into ACIR.
     #[error("Failed to compile circuit")]
-    CompilationError,
+    CompilationError(#[from] noirc_errors::Error),
```

Comment:
How does this work if a program returns more than one error?

Reply:
There will be problems with this, but maybe we can go and do something like this:
https://github.com/noir-lang/noir/blob/053dba0ccc18757aef8e12fa170103a46de07244/crates/noirc_frontend/src/hir/type_check/errors.rs#L41